sdk/storage/Azure.Storage.Common/tests/Shared/StorageTestBase.cs

```diff
@@ -155,7 +155,19 @@ private TenantConfiguration GetTestConfig(string name, Func<TenantConfiguration>
                 case RecordedTestMode.Playback:
                     if (!_playbackConfigCache.TryGetValue(name, out config))
                     {
-                        text = Recording.GetVariable(name, null);
+                        try
+                        {
+                            text = Recording.GetVariable(name, null);
+                        }
+                        catch
+                        {
+                            // If we can't find the variable, treat it as a mismatch recording.
+                            throw new TestRecordingMismatchException("no recordings found");
+                        }
+                        if (text == null)
+                        {
+                            throw new TestRecordingMismatchException($"Could not find variable '{name}'");
+                        }
```

Comment:
@kasobol-msft These changes should allow [RecordedTest] to work as expected. Previously this code would prevent it from re-recording.

Reply:
Shouldn't this go inside `Recording.GetVariable` ?

Reply:
This only affects test classes that do non-standard things with Recordings. We can move it if this becomes more common.

Reply:
How is attempting to retrieve variable from recording a non-standard thing?

Reply:
@pakrym what do you think ?

Reply:
The non-standard bit is custom managing how the recorded variable is set based on mode. This is usually done automatically by exposing the variable in the Environment and fetching it via `GetRecordedVariable` which automatically either gets it from the recording or your environment depending on mode.

Reply:
Perhaps, but I'd prefer not to take it on as part of this PR.

Reply:
I that case. I'd rather not include this fix in this PR at all and create issue to seek proper solution.
We already have enough debt in storage suite to deal with, let's not add more.